app/src/main/java/com/into/websoso/ui/main/MainActivity.kt

```
@@ -206,4 +214,25 @@ class MainActivity : BaseActivity<ActivityMainBinding>(activity_main) {
+    ) {
+        LIBRARY(menu_library),
+        HOME(menu_home),
+        EXPLORE(menu_explore),
+        FEED(menu_feed),
+        MY_PAGE(menu_my_page),
+        ;
+
+        companion object {
+            fun valueOf(id: Int): FragmentType =
+                entries.find { it.resId == id }
+                    ?: throw IllegalArgumentException()
```

detekt (1.23.8)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)